plugins/modules/host_template_info.py

```diff
@@ -0,0 +1,176 @@
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: host_template_info
+short_description: Retrieve details of host templates.
+description:
+  - Collects detailed information about individual or all host templates.
+  - The module supports C(check_mode).
+author:
+  - "Ronald Suplina (@rsuplina)"
+requirements:
+  - cm_client
+options:
+  cluster:
+    description:
+      - The associated cluster name.
+    type: str
+    required: yes
+    aliases:
+      - cluster_name
+  name:
+    description:
+      - The name of the host template.
+    type: str
+    required: no
+"""
+
+EXAMPLES = r"""
+---
+- name: Retrieve the defailts about a specific host template
+  cloudera.cluster.host_template_info
+    host: example.cloudera.com
+    username: "jane_smith"
+    password: "S&peR4Ec*re"
+    cluster: "cfm_cluster"
+    name: "cfm_host_template"
+
+- name: Retrieve the details about all host templates within the cluster
+  cloudera.cluster.host_template_info
+    host: example.cloudera.com
+    username: "jane_smith"
+    password: "S&peR4Ec*re"
+    cluster: "cfm_cluster"
+"""
+
+RETURN = r"""
+---
+host_template_info:
+  description:
+    - Details about host template.
+  type: list
+  elements: dict
+  returned: always
+  contains:
+    name:
+      description:
+        - The name of the host template
+      type: str
+      returned: always
+    cluster_name:
+      description: A reference to the enclosing cluster.
+      type: dict
+      returned: always
+    role_groups:
+      description:
+        - The names of the role config groups
+      type: list
+      returned: always
+"""
+
+
+class ClouderaHostTemplateInfo(ClouderaManagerModule):
+    def __init__(self, module):
+        super(ClouderaHostTemplateInfo, self).__init__(module)
+
+        # Set the parameters
+        self.cluster_name = self.get_param("cluster")
+        self.name = self.get_param("name")
+
+        # Initialize the return value
+        self.host_templates_output = []
+
+        # Execute the logic
+        self.process()
+
+    @ClouderaManagerModule.handle_process
+    def process(self):
+        try:
+            ClustersResourceApi(self.api_client).read_cluster(self.cluster_name)
+        except ApiException as ex:
+            if ex.status == 404:
+                self.module.fail_json(
+                    msg="Cluster does not exist: " + self.cluster_name
+                )
+            else:
+                raise ex
+
+        host_temp_api_instance = HostTemplatesResourceApi(self.api_client)
+        if self.name:
+            try:
+                self.host_templates_output = _parse_host_template_output(
+                    host_temp_api_instance.read_host_template(
+                        cluster_name=self.cluster_name,
+                        host_template_name=self.name,
+                    ).to_dict()
+                )
+            except ApiException as ex:
+                if ex.status == 404:
```

Comment:
If the host template does not exist, we shouldn't fail the module. We should return an empty set instead.  This isn't an exception, it's a known state -- it doesn't exist -- and a `fail_json` will halt a playbook (without some `block/rescue` business).  For example, the `service_info` module doesn't fail, it just reports an empty set: https://github.com/cloudera-labs/cloudera.cluster/blob/main/plugins/modules/service_info.py#L245-L257